### PR TITLE
Allow activating kokkos TPLS in Tribits, fix libdl issues

### PR DESCRIPTION
**Pull request by jjwilke**
@trilinos/kokkos 

## Motivation
Standalone Kokkos and Tribits do not currently have compatible TPL systems.
This attempts to bride the gap between them and adds the first "shared TPL: libdl
It also fixes with profiling tests that are now able to be activated in Trilinos.

## Testing
All unit tests in Kokkos will have profiling/libdl activated on them now.

### CODE REVIEW

**Comment by tasmith4** [comment]
@jjwilke I tried this out in my repo and got the following fatal error at configure-time:
```
CMake Error at packages/kokkos/cmake/kokkos_tpls.cmake:26 (MESSAGE):
  Enabled TPL CUDA inside TriBITS build, but this can only be enabled in a
  standalone build
```

**Comment by jjwilke** [comment]
@tasmith4 Sorry about that. I had local changes I didn't push. Please pull and try again.

**Comment by tasmith4** [comment]
@jjwilke it seems to pick up the Kokkos simple kernel timer now

**Comment by janciesko** [comment]
Ran into same issue, confirming that PR fixes this issue.

File: packages/kokkos/cmake/kokkos_tpls.cmake [hunk 1]
```diff
@@ -2,22 +2,44 @@ KOKKOS_CFG_DEPENDS(TPLS OPTIONS)
...
+  CMAKE_PARSE_ARGUMENTS(PARSED
+    ""
+    "TRIBITS"
+    ""
+    ${ARGN})
+
+  IF (PARSED_TRIBITS)
+    #this is also a TPL option you can activate with Tribits
+    #Tribits brings its own default
+    SET(DEFAULT TPL_ENABLE_${PARSED_TRIBTS})
```
Comment (ndellingwood) [inline packages/kokkos/cmake/kokkos_tpls.cmake:14]: @jjwilke typo
```suggestion
    SET(DEFAULT TPL_ENABLE_${PARSED_TRIBITS})
```

File: packages/kokkos/cmake/kokkos_tpls.cmake [hunk 2]
```diff
@@ -2,22 +2,44 @@ KOKKOS_CFG_DEPENDS(TPLS OPTIONS)
...
+      MESSAGE(FATAL_ERROR "Enabled TPL ${PKG} inside TriBITS build, "
+              "but this can only be enabled in a standalone build")
+  ENDIF()
 ENDFUNCTION()
 
 KOKKOS_TPL_OPTION(HWLOC   Off)
 KOKKOS_TPL_OPTION(LIBNUMA Off)
 KOKKOS_TPL_OPTION(MEMKIND Off)
-KOKKOS_TPL_OPTION(CUDA    Off)
+KOKKOS_TPL_OPTION(CUDA    Off TRIBITS CUDA)
```
Comment (ndellingwood) [inline packages/kokkos/cmake/kokkos_tpls.cmake:34]: Curious why the appended `TRIBITS CUDA` was needed and what issue it resolved (or prevented)?

**Review by ndellingwood — CHANGES_REQUESTED** [review]

**Review by ndellingwood — APPROVED** [review]

File: packages/kokkos/cmake/kokkos_tpls.cmake (hunk 1 above)
Comment (ndellingwood) [inline packages/kokkos/cmake/kokkos_tpls.cmake:14]: @jjwilke I think my typo fix here got wiped out
```suggestion
    SET(DEFAULT TPL_ENABLE_${PARSED_TRIBITS})
```

**Review by ndellingwood — CHANGES_REQUESTED** [review]

**Review by ndellingwood — APPROVED** [review]

**Comment by tasmith4** [comment]
@jjwilke this seemed like a useful fix for an important issue -- is there another PR to address this issue?

**Comment by ndellingwood** [comment]
@tasmith4 this PR was replaced by #8226, still work in progress

**Pull request closed without merging**